packages/components/src/disabled/test/index.tsx

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Internal dependencies
+ */
+import Disabled from '../';
+
+jest.mock( '@wordpress/dom', () => {
+	const focus = jest.requireActual( '../../../../dom/src' ).focus;
+	return {
+		focus: {
+			...focus,
+			focusable: {
+				...focus.focusable,
+				find( context: Element, options = { sequential: false } ) {
+					// In JSDOM, all elements have zero'd widths and height.
+					// This is a metric for focusable's `isVisible`, so find
+					// and apply an arbitrary non-zero width.
+					Array.from( context.querySelectorAll( '*' ) ).forEach(
+						( element ) => {
+							Object.defineProperties( element, {
+								offsetWidth: {
+									get: () => 1,
+								},
+							} );
+						}
+					);
+
+					return focus.focusable.find( context, options );
+				},
+			},
+		},
+	};
+} );
+
+describe( 'Disabled', () => {
+	const Form = () => (
+		<form>
+			<input />
+			<div contentEditable tabIndex={ 0 } />
+		</form>
+	);
+
+	it( 'will disable all fields', () => {
+		const { container } = render(
+			<Disabled>
+				<Form />
+			</Disabled>
+		);
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+		expect( input ).toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
+		expect( div ).not.toHaveAttribute( 'tabindex' );
+		expect( div ).not.toHaveAttribute( 'disabled' );
+	} );
+
+	it( 'should cleanly un-disable via reconciliation', () => {
+		const MaybeDisable = ( { isDisabled = true } ) =>
+			isDisabled ? (
+				<Disabled>
+					<Form />
+				</Disabled>
+			) : (
+				<Form />
+			);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+
+		expect( input ).not.toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
+		expect( div ).toHaveAttribute( 'tabindex' );
+	} );
+
+	it( 'will disable or enable descendant fields based on the isDisabled prop value', () => {
+		const MaybeDisable = ( { isDisabled = true } ) => (
+			<Disabled isDisabled={ isDisabled }>
+				<Form />
+			</Disabled>
+		);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+
+		expect( input ).toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
+
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		expect( input ).not.toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//  - it( 'will disable all fields on sneaky DOM manipulation' )
+	//
+	// Alas, JSDOM does not support MutationObserver:
+	//
+	//  https://github.com/jsdom/jsdom/issues/639
```

Comment:
We should look into adding these tests, since the `MutationObserver` is now supported — but that can definitely happen in a follow-up (if you have capacity to work on it!)